### PR TITLE
Eclipse  Type safety Java Error cleanup

### PR DESCRIPTION
**Pull request by jimzucker**
reference #343 - Type safety Java Error in eclipse (10 cases)
    There are a few cases left that need more research, the ones in the JIRA collectors broke tests when I fixed them.  The other 2 cases need more research on the proper resolution.

reference #352 - PMD Warnings - Avoid catching NullPointerException; consider removing the cause of the NPE - For now suppresed and added to TODO list
            @tabladrum noted that these should be fixed, for today I suppressed them and marked them for the TODO list to cleanup, in this way we can monitor PMD more effectively.

screenshot 2016-03-04 11 38 41

### CODE REVIEW

**Comment by jimzucker** [comment]
Looks like I missed a checkin, I will fix it.. give me a few minutes.

File: core/src/test/java/com/capitalone/dashboard/MarkdownTest.java [hunk 1]
```diff
@@ -60,9 +61,18 @@ private File getProjectRoot() {
...
+    	FileInputStream input = null;
+        try {
+        	input = new FileInputStream(filename);
+        	FileChannel channel = input.getChannel();
             ByteBuffer buffer = channel.map(FileChannel.MapMode.READ_ONLY, 0, (int) channel.size());
-            return Charset.forName("8859_1").newDecoder().decode(buffer);
+            CharBuffer result = Charset.forName("8859_1").newDecoder().decode(buffer);
+            input.close();
+			return result;
+        } catch (Exception e) {
```
Comment (jdamick) [inline core/src/test/java/com/capitalone/dashboard/MarkdownTest.java]: i dont understand why you changed this?

Comment (jimzucker) [inline core/src/test/java/com/capitalone/dashboard/MarkdownTest.java]: See #350  - Eclipse identified we where not close the FileChannel resource returned, so my goal was to fix that.

This was the original code:

    CharSequence fromFile(String filename) throws IOException {
        try (FileChannel channel = new FileInputStream(filename).getChannel()) {
            ByteBuffer buffer = channel.map(FileChannel.MapMode.READ_ONLY, 0, (int) channel.size());
            return Charset.forName("8859_1").newDecoder().decode(buffer);
        }
    }

**Comment by jimzucker** [comment]
@jdamick are you OK to accept this pull request now?

**Pull request merged**